### PR TITLE
Importing Burp Html reports

### PR DESCRIPTION
**Pull request by dormi**
### Spec
When uploading Burp XML files into Dradis, all of the request/response data is included. 
But in Html files, Burp only shows the relevant portion of the request/response data in their HTML reports , and the rest is hidden under `[SNIP]` portions.

This PR supports parsing Burp Html files so we can see those `snipped` request/response fields. 

### How to test

- Without using the the new plugin branch, upload a burp XML file. Then switch to the new branch and upload it again in another project (in the new branch, in the plugins dropdown you should see 2 different entries for burp::xml and burp::html). Compare both projects to check they look the same. 

- Upload the XML again using the command line in another project. The upload should still look the same (`thor dradis:plugins:burp:upload /path/to/burp_xml.xml`)

- Using Burp obtain a xml report and a html file for the same burp project.
   Upload the files in 2 different projects using the web UX: (same burp output exported as XML and  HTML):
Both projects should look exactly the same, with l few minor known differences: 
    - In the HTML one there is no IP in the Evidence, just the host name.
    - In the HTML one there is no Location field in the Evidence.
    - In the XML there is a "length" truncation error. In the HTML project that error is to present because the "request/response" fields have [SNIP] sections

- Using the command line upload a burp html file in a new project, and assert the project looks exactly the same as when we upload it using the web UX. (`PROJECT_ID=N thor dradis:plugins:burp:upload /path/to/burp_html.html`)

### CODE REVIEW

File: lib/dradis/plugins/burp/formats/html.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      link       = html_issue.first.css('a').attr('href').value
+      burp_id    = link[/\/(\d+)_.*/, 1]
+      issue_id   = html_issue.attr('id').value
+      issue_text =
+        template_service.process_template(
+          template: 'issue',
+          data: html_issue
+        )
+
+      logger.info { "Processing issue #{issue_id}" }
```
Comment (aapomm) [inline lib/dradis/plugins/burp/formats/html.rb]: Is it possible to use the issue title here? I just feel like the logs look arbitrary:
```
...
[12:35:39] Processing evidence 12.1

[12:35:41] Processing evidence 12.2

[12:36:27] Processing issue 13

[12:36:27] Processing evidence 13
...
```

File: lib/burp/html/issue.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
+module Burp
+  module Html
+  # We use this string to replace invalid UTF-8 bytes with.
+  # INVALID_UTF_REPLACE = '<?>'
```
Comment (aapomm) [inline lib/burp/html/issue.rb]: Remove unused code

File: lib/burp/html/issue.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        request_2: 'Request 2',
+        response_2: 'Response 2',
+        request_3: 'Request 3',
+        response_3: 'Response 3'
+      }
+
+      # look for the h2 headers in the html fragment
+      method_name = translations_table.fetch(method, method.to_s)
+      h2 = @html.xpath("//h2[text()='#{method_name}']").first
+      unless h2.nil?
```
Comment (aapomm) [inline lib/burp/html/issue.rb]: Is this equivalent to `if h2` which seems more readable?

File: lib/dradis/plugins/burp/formats/xml.rb [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      unless @hosts.include?(affected_host.label)
+        @hosts << affected_host.label
+        url = xml_issue.at('host').text
+        affected_host.set_property(:hostname, url)
+        affected_host.save
+      end
+
+      # Burp extensions don't follow the "unique type for every Issue" logic
+      # so we have to deal with them separately
+      if xml_issue.at('type').text.to_str == '134217728'
```
Comment (aapomm) [inline lib/dradis/plugins/burp/formats/xml.rb]: What does this number mean? I might be good to save this value in a semantic variable. Ex: `BURP_TYPE_ID = 134217728` (replace BURP_TYPE_ID with what it actually is)

File: lib/dradis/plugins/burp/gem_version.rb [hunk 5]
```diff
@@ -9,7 +9,7 @@ def self.gem_version
       module VERSION
         MAJOR = 3
         MINOR = 11
-        TINY = 0
```
Comment (aapomm) [inline lib/dradis/plugins/burp/gem_version.rb]: this seems like a big change and deserves a minor version bump to 3.12

Comment (dormi) [reply lib/dradis/plugins/burp/gem_version.rb]: Not sure about the version yet. 
If this is finished before `3.12`, may become a `3.11.1` ?

File: lib/tasks/thorfile.rb [hunk 6]
```diff
@@ -3,18 +3,27 @@ class BurpTasks < Thor
...
 
-    importer = Dradis::Plugins::Burp::Importer.new(task_options)
+    importer =
+      if File.extname(file_path) == '.xml'
+        Dradis::Plugins::Burp::Xml::Importer.new(task_options)
+      elsif File.extname(file_path) == '.html'
+        Dradis::Plugins::Burp::Html::Importer.new(task_options)
+      else
+        $stderr.puts "** Unsupported file. Must be .xml or .html"
+        exit(-1)
```
Comment (etdsoft) [inline lib/tasks/thorfile.rb]:
```suggestion
        exit(-2)
```

File: templates/html_evidence.fields [hunk 7]
```diff
@@ -0,0 +1,13 @@
+issue.host
```
Comment (etdsoft) [inline templates/html_evidence.fields]: shouldn't all of these be `html_evidence.` instead of `issue.`?

File: lib/burp/html/issue.rb [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    # Accepts a Nokogiri::XML::NodeSet
+    def initialize(html)
+      @html = Nokogiri::HTML(html.to_s)
+    end
+
+    # List of supported tags
+    def supported_tags
+      [
+        # tags with contents retrieved from inside the span header
+        :type, :name,
```
Comment (etdsoft) [inline lib/burp/html/issue.rb]: let's go with :alpha so it's a bit easier to find the symbol you're looking for

File: lib/burp/html/issue.rb [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        end
+
+        @summary
+      end
+    end
+
+    # List of supported tags to obtain from the summary html table
+    def summary_table_tags
+      [
+        :host, :severity, :path, :confidence
```
Comment (etdsoft) [inline lib/burp/html/issue.rb]: :alpha

File: lib/burp/xml/issue.rb [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    end
+
+    # List of supported tags. They can be attributes, simple descendants or
+    # collections (e.g. <references/>, <tags/>)
+    def supported_tags
+      [
+        # attributes
+
+        # simple tags
+        :serial_number, :type, :name, :host, :path, :location, :severity,
```
Comment (etdsoft) [inline lib/burp/xml/issue.rb]: it's a pitty to refactor w/o addressing :alpha!

File: lib/burp/xml/issue.rb [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+      # exception
+      unless supported_tags.include?(method)
+        super
+        return
+      end
+
+      # First we try the attributes. In Ruby we use snake_case, but in XML
+      # CamelCase is used for some attributes
+      translations_table = {
+        :background => 'issueBackground',
```
Comment (etdsoft) [inline lib/burp/xml/issue.rb]: new syntax + indenting?

File: lib/dradis-burp.rb [hunk 12]
```diff
@@ -6,3 +6,5 @@
 
 # Load supporting Burp classes
 require 'burp/issue'
+require 'burp/xml/issue'
```
Comment (etdsoft) [inline lib/dradis-burp.rb]: :alpha

File: templates/html_evidence.fields (hunk 7 above)
Comment (dormi) [reply templates/html_evidence.fields]: I am not sure, probably yes.
We are parsing a fragment of data that can contain both issue/evidence info.
So in some way it also makes sense to keep using issue, as we do in `evidence.fields` file.

**Pull request merged**